src/Symfony/Component/Workflow/Registry.php

```diff
@@ -23,19 +24,26 @@ class Registry
 
     /**
      * @param Workflow $workflow
-     * @param string   $className
+     * @param SupportStrategyInterface $supportStrategy
      */
-    public function add(Workflow $workflow, $className)
+    public function add(Workflow $workflow, SupportStrategyInterface $supportStrategy)
```

Comment:
Regardless of architecture (not familiar with the workings), you don't implement an interface here. This means you can safely remove the typehint and do a type check:
```
@param SupportStrategyInterface|string
public function add(Workflow $workflow, $supportStrategy)

if not instance of SupportStrategyInterface
    if not string
        throw invalid argument exception

    trigger deprecation, argument must be a SupportStrategyInterface as of 4.0
```